xarray/core/alignment.py

```diff
@@ -116,6 +116,136 @@ def align(
+    ...              coords={'lat': [35., 42.], 'lon': [100., 120.]})
+
+    >>> x
+    <xarray.DataArray (lat: 2, lon: 2)>
+    array([[25, 35],
+           [10, 24]])
+    Coordinates:
+    * lat      (lat) float64 35.0 40.0
+    * lon      (lon) float64 100.0 120.0
+
+    >>> y
+    <xarray.DataArray (lat: 2, lon: 2)>
+    array([[20,  5],
+           [ 7, 13]])
+    Coordinates:
+    * lat      (lat) float64 35.0 42.0
+    * lon      (lon) float64 100.0 120.0
+
+    >>> a, b = xr.align(x, y)
+    >>> a
+    <xarray.DataArray (lat: 1, lon: 2)>
+    array([[25, 35]])
+    Coordinates:
+    * lat      (lat) float64 35.0
+    * lon      (lon) float64 100.0 120.0
+    >>> b
+    <xarray.DataArray (lat: 1, lon: 2)>
+    array([[20,  5]])
+    Coordinates:
+    * lat      (lat) float64 35.0
+    * lon      (lon) float64 100.0 120.0
+
+    >>> a, b = xr.align(x, y, join='outer')
+    >>> a
+    <xarray.DataArray (lat: 3, lon: 2)>
+    array([[25., 35.],
+           [10., 24.],
+           [nan, nan]])
+    Coordinates:
+    * lat      (lat) float64 35.0 40.0 42.0
+    * lon      (lon) float64 100.0 120.0
+    >>> b
+    <xarray.DataArray (lat: 3, lon: 2)>
+    array([[20.,  5.],
+           [nan, nan],
+           [ 7., 13.]])
+    Coordinates:
+    * lat      (lat) float64 35.0 40.0 42.0
+    * lon      (lon) float64 100.0 120.0
+
+    >>> a, b = xr.align(x, y, join='outer', fill_value=-999)
+    >>> a
+    <xarray.DataArray (lat: 3, lon: 2)>
+    array([[  25,   35],
+           [  10,   24],
+           [-999, -999]])
+    Coordinates:
+    * lat      (lat) float64 35.0 40.0 42.0
+    * lon      (lon) float64 100.0 120.0
+    >>> b
+    <xarray.DataArray (lat: 3, lon: 2)>
+    array([[  20,    5],
+           [-999, -999],
+           [   7,   13]])
+    Coordinates:
+    * lat      (lat) float64 35.0 40.0 42.0
+    * lon      (lon) float64 100.0 120.0
+
+    >>> a, b = xr.align(x, y, join='left')
+    >>> a
+    <xarray.DataArray (lat: 2, lon: 2)>
+    array([[25, 35],
+           [10, 24]])
+    Coordinates:
+    * lat      (lat) float64 35.0 40.0
+    * lon      (lon) float64 100.0 120.0
+    >>> b
+    <xarray.DataArray (lat: 2, lon: 2)>
+    array([[20.,  5.],
+           [nan, nan]])
+    Coordinates:
+    * lat      (lat) float64 35.0 40.0
+    * lon      (lon) float64 100.0 120.0
+
+    >>> a, b = xr.align(x, y, join='right')
+    >>> a
+    <xarray.DataArray (lat: 2, lon: 2)>
+    array([[25., 35.],
+           [nan, nan]])
+    Coordinates:
+    * lat      (lat) float64 35.0 42.0
+    * lon      (lon) float64 100.0 120.0
+    >>> b
+    <xarray.DataArray (lat: 2, lon: 2)>
+    array([[20,  5],
+           [ 7, 13]])
+    Coordinates:
+    * lat      (lat) float64 35.0 42.0
+    * lon      (lon) float64 100.0 120.0
+
+    >>> a, b = xr.align(x, y, join='exact')
+    Traceback (most recent call last):
+    ...
+        "indexes along dimension {!r} are not equal".format(dim)
+    ValueError: indexes along dimension 'lat' are not equal
```

Comment:
This looks a bit awkward to me. How about just `ValueError: indexes along dimension 'lat' are not equal`? Or other ideas welcome / not terrible atm